### DISCUSSION TITLE
`npm run generate-docs` fails

### DISCUSSION THREAD

**Discussion by ReagentX** [saul/demofile#216] [Questions] [answered]
Christophers-MacBook-Pro:~ chris$ npm run generate-docs
    npm ERR! path /Users/chris/package.json
    npm ERR! code ENOENT
    npm ERR! errno -2
    npm ERR! syscall open
    npm ERR! enoent ENOENT: no such file or directory, open '/Users/chris/package.json'
    npm ERR! enoent This is related to npm not being able to find a file.
    npm ERR! enoent 

    npm ERR! A complete log of this run can be found in:
    npm ERR!     /Users/chris/.npm/_logs/2017-08-15T05_54_54_452Z-debug.log

**Comment by ReagentX** [comment]
0 info it worked if it ends with ok
    1 verbose cli [ '/usr/local/Cellar/node/8.3.0_1/bin/node',
    1 verbose cli   '/usr/local/bin/npm',
    1 verbose cli   'run',
    1 verbose cli   'generate-docs' ]
    2 info using npm@5.3.0
    3 info using node@v8.3.0
    4 verbose config Skipping project config: /Users/chris/.npmrc. (matches userconfig)
    5 verbose stack Error: ENOENT: no such file or directory, open '/Users/chris/package.json'
    6 verbose cwd /Users/chris
    7 verbose Darwin 16.7.0
    8 verbose argv "/usr/local/Cellar/node/8.3.0_1/bin/node" "/usr/local/bin/npm" "run" "generate-docs"
    9 verbose node v8.3.0
    10 verbose npm  v5.3.0
    11 error path /Users/chris/package.json
    12 error code ENOENT
    13 error errno -2
    14 error syscall open
    15 error enoent ENOENT: no such file or directory, open '/Users/chris/package.json'
    16 error enoent This is related to npm not being able to find a file.
    17 verbose exit [ -2, true ]

**Comment by ReagentX** [comment]
Also, this logged when I ran `npm install --save demo file`:

    npm WARN saveError ENOENT: no such file or directory, open '/Users/chris/package.json'
    npm notice created a lockfile as package-lock.json. You should commit this file.
    npm WARN enoent ENOENT: no such file or directory, open '/Users/chris/package.json'
    npm WARN chris No description
    npm WARN chris No repository field.
    npm WARN chris No README data
    npm WARN chris No license field.

**Comment by switz** [comment, marked as answer]
You have to clone the repo and `cd` into it. You're running all of these commands in your home directory `~`.

If you want to build the docs, you'll need the repo, but if you just want to use `demofile` run:

```
$ mkdir demotest
$ cd demotest
$ npm init # follow instructions
$ npm i demofile --save
```

Then create a file (index.js) and import the package `const demofile = require('demofile');`.

**Comment by saul** [comment]
Thanks @switz - that's correct.
